modules/apps/commerce/commerce-order-content-web/src/main/java/com/liferay/commerce/order/content/web/internal/health/status/CommerceGuestCheckoutAuthenticationCommerceHealthStatus.java

```diff
@@ -102,10 +101,11 @@ public void fixIssue(long companyId, long commerceChannelId)
 			privateLayout = false;
 		}
 
-		ServiceContext serviceContext =
-			ServiceContextThreadLocal.getServiceContext();
+		ServiceContext serviceContext = new ServiceContext();
 
+		serviceContext.setScopeGroupId(commerceChannel.getSiteGroupId());
 		serviceContext.setTimeZone(TimeZone.getDefault());
+		serviceContext.setUserId(PrincipalThreadLocal.getUserId());
```

Comment:
Isn't it better to use User currentUser = _userService.getCurrentUser(); ?
In this case it should be possibile to have an empty PrincipalThreadLocal, but you never know